src/rest.rs

```diff
@@ -776,19 +690,29 @@ impl Message {
     /// Encode the message ready to be sent in the body of a HTTP request.
     ///
     /// If the cipher is set, then use it to encrypt the message.
-    pub fn encode(&mut self, format: &Format, cipher: Option<&CipherParams>) -> Result<()> {
+    pub fn encode(&mut self, format: &Format, cipher: Option<&Cipher>) -> Result<()> {
+        let iv: IV = thread_rng().gen();
```

Comment:
It seems strange that the IV generation is up in the REST client, I think it belongs within the `encrypt` function.

Reply:
We need this for testing, it's not exposed in the public API though. What would you suggest instead? I'm against an extra field that's checked every time in production.

Reply:
I suggest it being a concern of the `encrypt` function, like it is currently: https://github.com/ably/ably-rust/blob/main/src/rest.rs#L357-L364

Reply:
Yes but that involves an extra field where the IV can be passed in so I don't really see it as that different. Except that field goes unused when not testing.

Reply:
One thing i I did consider is two functions, one with an iv and one that generates, would that work for you?

Reply:
Perhaps just keep the IV in the CipherParams with a builder method to set it, like currently?

Reply:
That still involves some extra code and runtime cost that's only there for testing. Doesn't really have a noticeable effect but doesn't sit well with me.

Reply:
But on the flip side, this introduces a runtime cost of generating a random IV even if the user isn't using encryption? Also, it's unclear how we can test that `Message.encode` encrypts a message as expected if it always generates a random IV?

Reply:
True, didn't consider encryption not being on. Will rethink.

Reply:
See last commit